server/protocols/jmap/src/test/java/org/apache/james/jmap/mailet/TextCalendarBodyToAttachmentTest.java

```diff
@@ -197,41 +150,57 @@ public void serviceShouldConvertTextBodyOfMessageToAttachmentWhenTextCalendar()
     }
 
     @Test
-    public void serviceShouldKeepAllMessageHeaderWhenConverting() throws Exception {
-        MimeMessage message = MimeMessageBuilder.mimeMessageFromStream(new ByteArrayInputStream(MESSAGE_CONTENT.getBytes()));
-        List<Header> allHeaders = Collections.list(message.getAllHeaders());
-        List<String> allStringHeaders = allHeaders.stream()
-            .map(Header::getName)
-            .collect(Guavate.toImmutableList());
+    public void serviceShouldKeepHeaderValueExceptContentTypeAndMessageIdAndEncodingWhenConverting() throws Exception {
+        List<String> headerChanged = ImmutableList.of("Content-Type", "Message-ID", "Content-transfer-encoding");
+        List<Header> allHeaders = Collections.list(calendarMessage.getAllHeaders());
 
         Mail mail = FakeMail.builder()
-            .mimeMessage(message)
+            .mimeMessage(calendarMessage)
             .build();
 
         mailet.service(mail);
 
         List<Header> actualHeaders = Collections.list(mail.getMessage().getAllHeaders());
-        List<String> actualStringHeaders = actualHeaders
-            .stream()
-            .map(Header::getName)
-            .collect(Guavate.toImmutableList());
+        for (Header header : allHeaders) {
+            if (!headerChanged.contains(header.getName())) {
+                assertThat(actualHeaders)
+                    .extracting(Header::getName, Header::getValue)
+                    .contains(Tuple.tuple(header.getName(), header.getValue()));
+            }
+        }
+    }
+
+    @Test
+    public void contentTypeOfAttachmentShouldBeTakenFromOriginalMessage() throws Exception {
+        String originalContentType = calendarMessage.getContentType();
```

Comment:
What is the value of originalContentType?

String originalContentType = "text/calendar; method=REPLY; charset=UTF-8"

Overwize we don't knwo what you assert!